uint8 state
---
bool success
